src/main/java/io/quarkus/security/identity/SecurityIdentity.java

```diff
@@ -62,16 +63,21 @@ default <T extends Principal> T getPrincipal(Class<T> clazz) {
 
     /**
      * Checks if a user has a given role. These roles must be resolvable in advance for every request.
-     * <p>
-     * If more advanced authorization support is required than can be provided by a simple role based system
-     * then {@link #checkPermission(Permission)} and {@link #checkPermissionBlocking(Permission)} should be used
-     * instead.
-     * <p>
      *
      * @return <code>true</code> if the identity has the specified role.
      */
     boolean hasRole(String role);
 
+    /**
+     * Returns the set of all permissions held by the user. These permissions must be resolvable in advance for every request.
+     * <p>
+     * This set should either be unmodifiable, or a defensive copy so attempts to change the permission set do not modify
+     * the underlying identity.
+     *
+     * @return The set of all permissions held by the user
```

Comment:
You see, when you say `set of all permissions held by the user` it is true, but users might not realize that `checkPermission` checks also functions. for example, keycloak authorized extension is adding such a checker.

Reply:
@michalvavrik `the user` is referenced in other parts of this API which is why I used it, it is really `set of all permissions held by the current SecurityIdentity`